elements/lisk-p2p/test/integration/p2p.ts

```diff
@@ -1475,4 +1475,82 @@ describe('Integration tests for P2P library', () => {
 			});
 		});
 	});
+
+	describe('Network with peer inbound eviction protection enabled', () => {
+		const NETWORK_PEER_COUNT_WITH_LIMIT = 10;
+		const MAX_INBOUND_CONNECTIONS = 5;
+		const POPULATOR_INTERVAL_WITH_LIMIT = 10;
+		beforeEach(async () => {
+			p2pNodeList = [...new Array(NETWORK_PEER_COUNT_WITH_LIMIT).keys()].map(
+				index => {
+					// Each node will have the previous node in the sequence as a seed peer except the first node.
+					const seedPeers = [
+						{
+							ipAddress: '127.0.0.1',
+							wsPort:
+								NETWORK_START_PORT +
+								((index + 1) % NETWORK_PEER_COUNT_WITH_LIMIT),
+						},
+					];
+
+					const nodePort = NETWORK_START_PORT + index;
+					return new P2P({
+						connectTimeout: 300,
+						ackTimeout: 5000,
+						seedPeers,
+						wsEngine: 'ws',
+						populatorInterval: POPULATOR_INTERVAL_WITH_LIMIT,
+						maxOutboundConnections: MAX_INBOUND_CONNECTIONS,
+						maxInboundConnections: 5,
+						evictionProtectionEnabled: true,
+						nodeInfo: {
+							wsPort: nodePort,
+							nethash:
+								'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
+							version: '1.0.1',
+							protocolVersion: '1.0.1',
+							minVersion: '1.0.0',
+							os: platform(),
+							height: 0,
+							broadhash:
+								'2768b267ae621a9ed3b3034e2e8a1bed40895c621bbb1bbd613d92b9d24e54b5',
+							nonce: `O2wTkjqplHII${nodePort}`,
+						},
+					});
+				},
+			);
+
+			// Start nodes incrementally to make inbound eviction behavior predictable
+			p2pNodeList.forEach(async p2p => {
+				await wait(600);
+				p2p.start();
+			});
+			await wait(1000);
+		});
+
+		afterEach(async () => {
+			await Promise.all(
+				p2pNodeList
+					.filter(p2p => p2p.isActive)
+					.map(async p2p => await p2p.stop()),
+			);
+			await wait(100);
+		});
+
+		describe('Inbound peer evictions', () => {
+			it('should evict inbound peer which connected the latest', async () => {
+				p2pNodeList.forEach(async (p2p, i) => {
```

Comment:
The test fails. Careful with the `async` inside a forEach.
```suggestion
				p2pNodeList.forEach((p2p, i) => {
```